### PR TITLE
Add equalto

### PR DESCRIPTION
**Pull request by ararslan**

### CODE REVIEW

**Comment by ararslan** [comment]
Hm, 0.5 doesn't like something about the tests I copied from Base.

File: src/Compat.jl [hunk 1]
```diff
@@ -663,6 +663,30 @@ end
...
+
+                EqualTo(x::T) where {T} = new{T}(x)
+            end
+        """)
+    else
+        include_string(@__MODULE__, """
+            immutable EqualTo{T} <: Function
+                x::T
+
+                EqualTo{T}(x::T) = new(x)
```
Comment (yuyichao) [inline src/Compat.jl]: This defines a method for `EqualTo{T}` on 0.5.

File: test/runtests.jl [hunk 2]
```diff
@@ -823,6 +823,17 @@ end
 # 0.7
 @test isconcrete(Int)
 
+# 0.7
+let a = [0,1,2,3,0,1,2,3]
+    @test findfirst(equalto(3), [1,2,4,1,2,3,4]) == 6
+    @test findfirst(!equalto(1), [1,2,4,1,2,3,4]) == 2
```
Comment (yuyichao) [inline test/runtests.jl]: This probably won't work on 0.5 either.

Comment (ararslan) [reply test/runtests.jl]: It does--there's a Compat for it :)

**Pull request merged**